staging/src/k8s.io/client-go/util/certificate/csr/csr.go

```diff
@@ -46,16 +46,17 @@ import (
 // status, once approved by API server, it will return the API server's issued
 // certificate (pem-encoded). If there is any errors, or the watch timeouts, it
 // will return an error.
-func RequestCertificate(client certificatesclient.CertificateSigningRequestInterface, csrData []byte, name string, usages []certificates.KeyUsage, privateKey interface{}) (req *certificates.CertificateSigningRequest, err error) {
+func RequestCertificate(client certificatesclient.CertificateSigningRequestInterface, csrData []byte, name string, signerName string, usages []certificates.KeyUsage, privateKey interface{}) (req *certificates.CertificateSigningRequest, err error) {
```

Comment:
update ensureCompatible() to make sure that if `new.Spec.SignerName` and `orig.Spec.SignerName` are both non-nil, they match (and add unit tests for the nil, matching, and mismatching cases)